src/main/java/com/google/api/codegen/transformer/SurfaceNamer.java

```diff
@@ -812,14 +813,19 @@ public String getSampleAppClassName() {
    * The type name of the Grpc service class This needs to match what Grpc generates for the
    * particular language.
    */
-  public String getGrpcServiceClassName(InterfaceModel apiInterface) {
+  public String getRpcServiceClassName(
+      InterfaceModel apiInterface, GapicProductConfig productConfig) {
     NamePath namePath =
-        typeNameConverter.getNamePath(getModelTypeFormatter().getFullNameFor(apiInterface));
-    String grpcContainerName =
-        publicClassName(Name.upperCamelKeepUpperAcronyms(namePath.getHead(), "Grpc"));
+        typeNameConverter.getNamePath(getTypeFormatter().getFullNameFor(apiInterface));
+    String rpcContainerName =
+        publicClassName(
+            Name.upperCamelKeepUpperAcronyms(
+                namePath.getHead(),
+                Name.from(productConfig.getTransportProtocol().name().toLowerCase())
+                    .toUpperCamel()));
```

Comment:
The original function is very tied to grpc, because it is trying to generate the same names that grpc generates - it feels odd to generalize it.

Reply:
Ok, reverted back to specifically for grpc use.